release/config_release.py

```diff
@@ -0,0 +1,372 @@
+
+  Args:
+    config_group: ConfigGroup, the ConfigGroup to generate configs for.
+    bazel_version: string, the version of Bazel used to generate the configs.
+
+  Returns:
+    None
+  """
+  with open(config_group.get_toolchain_bazelrc_path(),
+            "w") as toolchain_bazelrc_file:
+    # Create or update toolchain.bazelrc file.
+    with open(_get_tpl("toolchain.bazelrc.tpl"), "r") as tpl_file:
+      tpl = Template(tpl_file.read()).substitute(
+          CONFIG_VERSION=config_group.version,
+          BAZEL_VERSION=bazel_version,
+          PACKAGE=config_group.package,
+          PLATFORM=config_group.platform_target,
+      )
+
+      toolchain_bazelrc_file.write(tpl)
+
+
+def _create_bazelrc_and_update_link(bazel_version):
+  """Create new sample .bazelrc file and update latest.bazelrc symlink.
+
+  Args:
+    bazel_version: string, the version of Bazel used to generate the configs.
+
+  Returns:
+    None
+  """
+  bazelrc_path = os.path.join(
+      BAZELRC_DIR, "bazel-{version}.bazelrc".format(version=bazel_version))
+
+  # Remove old version of this .bazelrc file.
+  if os.path.exists(bazelrc_path):
+    os.remove(bazelrc_path)
+
+  with open(bazelrc_path, "w") as bazelrc_file:
+    # Write license header.
+    with open(_get_tpl("license.tpl"), "r") as license_header:
+      bazelrc_file.write(license_header.read())
+
+    # Write sample .bazelrc body.
+    with open(_get_tpl("bazelrc.tpl"), "r") as tpl_file:
+      tpl = Template(tpl_file.read()).substitute(BAZEL_VERSION=bazel_version)
+      bazelrc_file.write(tpl)
+
+  # Update latest.bazelrc link
+  if os.path.exists(LATEST_BAZELRC_LINK):
+    os.remove(LATEST_BAZELRC_LINK)
+  os.symlink(os.path.basename(bazelrc_path), LATEST_BAZELRC_LINK)
+
+
+def _create_target_if_not_exists(config_group, config, bazel_version, sha,
+                                 target, date):
+  """Create the new docker_toolchain_autoconfig target if not exists.
+
+  Args:
+    config_group: ConfigGroup, the ConfigGroup to generate configs for.
+    config: Config, the Config to generate configs for.
+    bazel_version: string, the version of Bazel used to generate the configs.
+    sha: string, SHA256 of the container used to generate the configs.
+    target: string, the base name of the docker_toolchain_autoconfig target.
+    date: string, the date this script is executed.
+
+  Returns:
+    None
+  """
+  with open(config_group.get_target_build_path(), "a+") as build_file:
+    if target not in build_file.read():
+      with open(_get_tpl("%s.tpl" % config.config_type), "r") as tpl_file:
+        tpl = Template(tpl_file.read()).substitute(
+            DATE=date,
+            DISTRO=config_group.distro,
+            CONFIG_VERSION=config_group.version,
+            BAZEL_VERSION=bazel_version,
+            CONTAINER_NAME=config_group.container,
+            CONTAINER_SHA=sha)
+
+        build_file.write(tpl)
+
+
+def _execute_and_extract_configs(config_group, config, target):
```

Comment:
Consider if it would be good to split up data (the config_groups) from two types of processing (1. creating build and bazel rc files and 2. running targets to generate configs) (into separate py files, but again, I'm not python expert, so not sure what the best way to split is).

Reply:
Separate functions into different files and modules. PTAL.